src/cmake/externalpackages.cmake

```diff
@@ -57,7 +57,6 @@ checked_find_package (Imath REQUIRED
 
 checked_find_package (OpenEXR REQUIRED
     VERSION_MIN 3.1
-    NO_FP_RANGE_CHECK
```

Comment:
Why? I must have had that in for a reason (though now I'm not sure what problem it solved).

Reply:
I think I see what's happening. I ran into this problem, as I currently have another version of `OpenEXR 2.5.7` installed on the system, which is too old compared to the one specified by `checked_find_package()` -[ which is 3.1](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/db9fe349d007feeceb25f3b6fec8f12ff54d6ea3/src/cmake/externalpackages.cmake#L59).

With  `NO_FP_RANGE_CHECK` enabled, we should handle and always update [_pkg_VERSION_MIN](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/db9fe349d007feeceb25f3b6fec8f12ff54d6ea3/src/cmake/dependency_utils.cmake#L352) as well in order to set version range to the maximum range available (ie 0.0.1). Currently we don't do that which puts `2.5.7` outside of the valid range - which triggers a local build for version `3.x` .

Now, with two versions available the '[REFIND](https://github.com/AcademySoftwareFoundation/OpenImageIO/blob/db9fe349d007feeceb25f3b6fec8f12ff54d6ea3/src/cmake/dependency_utils.cmake#L429)' step does not enforce which version `find_package()` should select. It happens (at least in my case) that it picks the original one and not the one just built.
So in other words I'm wondering if we should also specify  `${${pkgname}_REFIND_VERSION} REQUIRED` to the `find_package()` in order to enforce that?

Reply:
I remember why the NO_FP_RANGE_CHECK now. So just as background, this is a signal to checked_find_package to NOT pass a version number (or range) to the underlying call to find_package. Now, why would we want to do that?

The dependency's exported BlahConfig.cmake/etc configuration will verify its declared version against that being demanded by the find_package call. There are a few methods that can be specified to write_basic_package_version_file when you have it create the BlahConfigVersion: AnyNewerVersion, SameMajorVersion, SameMinorVersion, ExactVersion.

The problem is that most packages export using SameMajorVersion. And that's probably the right choice for them, in the sense that the authors of a package such as OpenEXR know they have good source compatibility within a major version, but not between different major versions.

But we often want to make OIIO able to accept quite a wide range of possible versions of a dependency -- including spanning across major version boundaries, which we handle explicitly through the necessary `#if` statements and whatnot so that we really are able to use either major version. So what do we ask for? If we ask for OpenEXR 2.5 but find 3.1, we don't want the rejection we would get with OpenEXR's declaration that it's not compatible across major versions. 

This is partly due to kind of a misunderstanding. I think that when people write `find_package (Mypackage 2.5)` they probably *mean* "a minimum of 2.5", and don't realize that this won't work with 3.1 unless the upstream package said AnyNewerVersion. You just can't override the matching rules on the downstream/requesting end.

So to get around it, we use NO_FP_RANGE_CHECK to suppress our declaring the range at all, so it's not rejected by find_package. We do our own range check instead.




The problem is that with

Reply:
I see, thanks for clarification.
One more question regarding the 'Refind' step. If during the build we determine to build local version of the library, as the existing one (found on the system) is too old. Should the `find_package` in the 'Refind' step actually only search for the local version. I'm not sure in what order CMake performs the scan but it feels there's a chance to pick up the version we don't want to use, no?

Reply:
Yes, if we build our own, we should try hard to ensure that the re-find only can get the one we just built.

I think we're doing great when the dependency isn't found at all and we build it.

But I've had some trouble in the past with the case where it's found but too old. So it's possible that we are still missing the right cmake arguments somewhere to 100% ensure that we can't pick up the old one.

Reply:
Specifically, we want the re-find to be rigged to ONLY look in the `${PROJECT_NAME}_LOCAL_DEPS_ROOT` destination where we put the auto-built version, and not in the system areas, and also we probably want the include area under there to absolutely be listed before any system include areas so that we don't accidentally pick up the system (wrong version) headers for anything we tried to build locally.

Reply:
I wasn't able to post this earlier but I can reproduce the 'Refind' related problem mentioned above. Here is the annotated output without the current changes:
```
make OpenImageIO_BUILD_MISSING_DEPS=all

-- Found Imath 3.1.10
-- OpenEXR  is outside the required range 3.1...10000.0.0
-- Building package OpenEXR  locally  <-------------------- Build our own version
--         Building local OpenEXR 3.2.4 from https://github.com/AcademySoftwareFoundation/OpenEXR
--         Downloading local https://github.com/AcademySoftwareFoundation/OpenEXR
HEAD is now at a1a00ff Add PR 1674 to notes
-- Configuring done (0.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/kuba/SRC/OIIO/oiio-kaarrot/build/deps/OpenEXR-build
gmake[1]: Entering directory '/home/kuba/SRC/OIIO/oiio-kaarrot/build/deps/OpenEXR-build'
...
...
...
-- Up-to-date: /home/kuba/SRC/OIIO/oiio-kaarrot/build/deps/dist/include/OpenEXR/ImfUtilExport.h
gmake[1]: Leaving directory '/home/kuba/SRC/OIIO/oiio-kaarrot/build/deps/OpenEXR-build'
-- Refinding OpenEXR with OpenEXR_ROOT=/home/kuba/SRC/OIIO/oiio-kaarrot/build/deps/dist
-- Found OpenEXR 2.5.7    <-------------------- Still selects system library 
-- Found libjpeg-turbo 3.0.4
```
Passing  `${${pkgname}_REFIND_VERSION} REQUIRED` addresses that.